src/components/RedeemCouponPopup.tsx

```diff
@@ -1,38 +1,135 @@
+  align-items: center;
+`
+
+const StyledCloseButton = styled(CloseButton)`
+  position: absolute;
+  top: 20px;
+  right: 20px;
+`
+
+const StyledRedeemContainer = styled.section`
+  width: 50%;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  gap: 20px;
+
+  @media ${({ theme }) => theme.breakpoints.mobile} {
+    width: 90%;
+  }
+`
+
+const StyledInputContainer = styled.div`
+  width: 100%;
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+  gap: 10px;
+`
+
+const StyledInput = styled(RegularInput)`
+  width: 100%;
+  border: 2px solid ${({ theme }) => theme.colors.otterBlack};
+  border-radius: 10px;
+  padding: 20px;
+
+  ::placeholder {
+    color: ${({ theme }) => theme.colors.lightGray400};
+    opacity: 1;
+  }
+`
 
 enum State {
+  QueryAmount,
   Loading,
   Success,
 }
 
 interface Props {
   coupon: ItemMetadata
+  item: Item | undefined
   onClose: () => void
 }
 
 export default memo(function RedeemCouponPopup({
   coupon: { tokenId: id, productType, productImages },
+  item,
   onClose,
 }: Props) {
-  const [state, setState] = useState<State>(State.Loading)
+  const [state, setState] = useState<State>(State.QueryAmount)
+  const [amount, setAmount] = useState(1)
   const { resetRedeem, redeem, redeemState } = useRedeemProduct()
+  const onRedeem = () => {
+    redeem(id, amount)
+    setState(State.Loading)
+  }
   useEffect(() => {
-    redeem(id)
+    if (!item) {
+      redeem(id, 1)
+      setState(State.Loading)
+    }
   }, [])
   useEffect(() => {
     if (redeemState.state === 'Success') {
       setState(State.Success)
     } else if (redeemState.state === 'Fail' || redeemState.state === 'Exception') {
       alert(redeemState.status.errorMessage || '')
       resetRedeem()
-      onClose()
+      setState(State.QueryAmount)
     }
   }, [redeemState])
+  const submitted = state !== State.QueryAmount
+  const { t } = useTranslation('', { keyPrefix: 'my_items' })
+  const maxAmount = item ? item.amount : 1
 
+  if (state === State.QueryAmount)
+    return (
+      <Fullscreen show>
+        <StyledRedeemCouponPopup>
+          <StyledCloseButton onClose={onClose} />
+          <StyledRedeemContainer>
+            <Headline>{t('open')}</Headline>
+            <ItemCell showDetailsPopup hideAmount item={item} size={240} />
+            <StyledInputContainer>
+              <ContentSmall>{t('transfer.amount')}</ContentSmall>
+              <StyledInput
+                disabled={submitted}
+                type="number"
+                max={maxAmount}
+                min={1}
+                value={amount}
+                onChange={e => setAmount(Number(e.target.value))}
+              />
+            </StyledInputContainer>
+            <Button Typography={Headline} width="100%" loading={submitted} onClick={onRedeem}>
+              {t('open')}
+            </Button>
+          </StyledRedeemContainer>
+        </StyledRedeemCouponPopup>
+      </Fullscreen>
```

Comment:
Please extract this part of code into a component.

Reply:
Done, check to see if this is this is what you have in mind